macros/src/types/named.rs

```diff
@@ -92,22 +93,33 @@ fn format_field(
         return Ok(());
     }
 
+    if type_as.is_some() && type_override.is_some() {
+        syn_err!("`type` is not compatible with `as`")
+    }
+
+    let parsed_ty = if let Some(ref type_as) = type_as {
+        syn::parse_str::<Type>(type_as)?
+    } else {
+        field.ty.clone()
+    };
+
     let (ty, optional_annotation) = match optional {
         Optional { optional: true, nullable } => {
             let inner_type = extract_option_argument(&field.ty)?; // inner type of the optional
```

Comment:
I *think* we wanna change this to `let inner_type = extract_option_argument(&parsed_ty)?`.
To get `as` working with `optional` then, you do 
```rust
#[derive(TS)]
struct Outer {
    #[ts(as = "Option<ExternalTypeDef>")]
    #[ts(optional)]
    x: Unsupported
}
```

The alteratives I can think off seem unintuitive, from a users perspective. Like this, the `as` override just.. overrides the field, even if it's an option.

Reply:
With just that change, I think everything works like a charm:
```rust
#[derive(TS)]
struct Outer {
    #[ts(as = "Option<ExternalTypeDef>")]
    #[ts(optional = nullable, inline)]
    x: Unsupported
}
```
```ts
type Outer = { x?: { a: number, b: number, c: number, } | null, }
```

Reply:
I've got a commit ready if you think this makes sense @escritorio-gustavo

Reply:
I think I get it, but how does `#[ts(optional, as = "...")]` (without nullable) behave? This codepath was ignoring `as`

Reply:
Nevermind I just realized `inner_type` is the variable used in the codepath I mentioned

Reply:
> I've got a commit ready if you think this makes sense @escritorio-gustavo

Seems good to me!